service/src/test/java/bio/terra/pipelines/service/JobsServiceTest.java

```diff
@@ -0,0 +1,69 @@
+package bio.terra.pipelines.service;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.ArgumentMatchers.argThat;
+import static org.mockito.Mockito.*;
+
+import bio.terra.pipelines.db.JobsDao;
+import bio.terra.pipelines.db.exception.DuplicateObjectException;
+import bio.terra.pipelines.service.model.Job;
+import bio.terra.pipelines.testutils.BaseUnitTest;
+import java.util.UUID;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.mock.mockito.MockBean;
+
+class JobsServiceTest extends BaseUnitTest {
+
+  @Autowired private JobsService jobsService;
+  @MockBean private JobsDao jobsDao;
+
+  // parameters used repeatedly by various tests, and things we'll want to mocks to respond to
+  // universally
+  private final String testUserId = "testUser";
+  private final String testGoodPipelineId = "testGoodPipeline";
+  private final String testPipelineVersion = "testPipelineVersion";
+
+  // We'll need these to configure the dao to return selectively good or bad values
+  private final UUID testGoodUUID = UUID.randomUUID();
+  private final UUID testBadUUID = UUID.randomUUID();
+
+  @BeforeEach
+  void initMocks() {
+    // dao throws on job containing bad id and returns good uuid on job containing good uuid
+    when(jobsDao.createJob(argThat((Job j) -> j.getJobId() == testBadUUID)))
+        .thenThrow(DuplicateObjectException.class);
```

Comment:
`jobsDao.createJob()` now returns `null` if a duplicate is found - I think we need to update this logic and the `testCreateJob_unsuccessfulWriteDaoThrows()` test

Reply:
I believe it doesn't behave that way yet (relative to the current state of merged code in main).  Is TSPS-19 ready to be reviewed and merged yet?  If not, we can update the tests when it is.  Otherwise, I'll have to check in tests that are currently broken.